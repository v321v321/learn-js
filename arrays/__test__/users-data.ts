export interface User {
  name: string;
  experience: number;
}

export const usersData = [
  { name: 'Zemlak', experience: 14 },
  { name: 'Upton', experience: 13 },
  { name: 'Wisoky', experience: 3 },
  { name: 'Balistreri', experience: 8 },
  { name: 'Braun', experience: 13 },
  { name: 'Walsh', experience: 9 },
  { name: 'Cummings', experience: 3 },
  { name: 'Johnston', experience: 8 },
  { name: 'Dickens', experience: 14 },
  { name: 'Treutel', experience: 6 },
  { name: 'Buckridge', experience: 12 },
  { name: 'Veum', experience: 13 },
  { name: 'Turcotte', experience: 10 },
  { name: 'Bartell', experience: 8 },
  { name: 'Gottlieb', experience: 7 },
  { name: 'Will', experience: 15 },
  { name: 'Gleason', experience: 14 },
  { name: 'Nikolaus', experience: 8 },
  { name: 'Durgan', experience: 4 },
  { name: 'Mayert', experience: 14 },
  { name: 'Flatley', experience: 8 },
  { name: 'Koelpin', experience: 3 },
  { name: 'Feest', experience: 14 },
  { name: 'Thiel', experience: 10 },
  { name: 'Collins', experience: 8 },
  { name: 'Hudson', experience: 6 },
  { name: 'Crona', experience: 7 },
  { name: 'Kihn', experience: 13 },
  { name: 'Murray', experience: 8 },
  { name: 'Rempel', experience: 2 },
  { name: 'Gleichner', experience: 6 },
  { name: 'Keebler', experience: 10 },
  { name: 'Abernathy', experience: 4 },
  { name: 'Wolf', experience: 4 },
  { name: 'Fadel', experience: 11 },
  { name: 'Simonis', experience: 15 },
  { name: 'Ullrich', experience: 7 },
  { name: 'Jones', experience: 5 },
  { name: 'Langworth', experience: 4 },
  { name: 'Lind', experience: 14 },
  { name: 'Kuphal', experience: 9 },
  { name: 'Jerde', experience: 4 },
  { name: 'Fritsch', experience: 4 },
  { name: 'Stanton', experience: 3 },
  { name: 'Turner', experience: 9 },
  { name: 'Nolan', experience: 9 },
  { name: 'Gerlach', experience: 3 },
  { name: 'Boyle', experience: 10 },
  { name: 'Pfannerstill', experience: 6 },
  { name: 'Cormier', experience: 9 },
  { name: 'Padberg', experience: 12 },
  { name: 'Purdy', experience: 13 },
  { name: 'Ritchie', experience: 4 },
  { name: 'Doyle', experience: 12 },
  { name: 'Reichert', experience: 13 },
  { name: 'Breitenberg', experience: 13 },
  { name: 'Erdman', experience: 11 },
  { name: 'Schumm', experience: 9 },
  { name: 'Pfeffer', experience: 3 },
  { name: 'VonRueden', experience: 5 },
  { name: 'Wisozk', experience: 10 },
  { name: 'Metz', experience: 8 },
  { name: 'Barton', experience: 10 },
  { name: 'Ondricka', experience: 9 },
  { name: 'Corkery', experience: 14 },
  { name: 'Okuneva', experience: 13 },
  { name: 'Brown', experience: 13 },
  { name: 'Jacobson', experience: 12 },
  { name: 'Hills', experience: 5 },
  { name: 'Lueilwitz', experience: 4 },
  { name: 'Ziemann', experience: 10 },
  { name: 'Beier', experience: 8 },
  { name: 'Lemke', experience: 5 },
  { name: 'Altenwerth', experience: 2 },
  { name: 'Keeling', experience: 7 },
  { name: 'Crist', experience: 4 },
  { name: 'Deckow', experience: 3 },
  { name: 'Pollich', experience: 6 },
  { name: 'Gusikowski', experience: 7 },
  { name: 'Stehr', experience: 5 },
  { name: 'Jast', experience: 9 },
  { name: 'Nicolas', experience: 4 },
  { name: 'Hermiston', experience: 2 },
  { name: 'Moore', experience: 12 },
  { name: 'Kautzer', experience: 2 },
  { name: 'Huel', experience: 15 },
  { name: 'Heidenreich', experience: 8 },
  { name: 'Conn', experience: 4 },
];
