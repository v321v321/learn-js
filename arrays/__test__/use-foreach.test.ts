import { expect, test } from 'vitest';
import {
  getUsersWithMinExperience,
  getUsersWithNameStartWithS,
} from '../use-foreach';
import { usersData } from './users-data';

test('Get users with min experience', () => {
  expect(getUsersWithMinExperience(usersData)).toMatchSnapshot();
});

test('Get users count with name starts with "S"', () => {
  expect(getUsersWithNameStartWithS(usersData)).toMatchSnapshot();
});
