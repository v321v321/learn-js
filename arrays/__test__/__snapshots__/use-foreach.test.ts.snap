// Vitest Snapshot v1

exports[`Get users count with name starts with "S" 1`] = `4`;

exports[`Get users with min experience 1`] = `
[
  {
    "experience": 2,
    "name": "Rempel",
  },
  {
    "experience": 2,
    "name": "Altenwerth",
  },
  {
    "experience": 2,
    "name": "Hermiston",
  },
  {
    "experience": 2,
    "name": "Kautzer",
  },
]
`;
